### PR TITLE
Replace TravisCI with AppVeyor for Linux builds

### PR DESCRIPTION
**Pull request by merbla**
**What issue does this PR address?**
* Removes the need for two build systems.
* Removes TravisCI builds and replaces it with a Linux (Ubuntu) build in the AppVeyor matrix.

**Does this PR introduce a breaking change?**
* No

**Please check if the PR fulfills these requirements**
- [x] The commit follows our [guidelines](https://github.com/serilog/serilog/blob/dev/CONTRIBUTING.md)
- [x] Unit Tests for the changes have been added (for bug fixes / features)

**Other information**:
* https://www.appveyor.com/docs/build-configuration/#specializing-matrix-job-configuration
* https://www.appveyor.com/blog/2018/04/25/specialized-build-matrix-configuration-in-appveyor/

### CODE REVIEW

**Comment by adamchester** [comment]
Do we still want to publish the artifacts from the linux build, even if we don't publish them beyond the appveyor tab? If not, is there some way to remove the warning? I'm ok with leaving it as-is too. 

https://ci.appveyor.com/project/serilog/serilog/builds/23592659/job/rpdjphu4hqmlqqx0#L220

**Comment by adamchester** [comment]
As a quick sanity check, I had a quick at which tests were running in both builds. 

In windows builds there are [570 tests passed](https://ci.appveyor.com/project/serilog/serilog/builds/23592659/job/jyquhip8r32dmws3#L223) for `netcoreapp2.0`.

In the linux build, there is also [570 tests passed](https://ci.appveyor.com/project/serilog/serilog/builds/23592659/job/rpdjphu4hqmlqqx0#L203) for `netcoreapp2.0`.

Only `netcoreapp2.0` tests run on both platforms.
👍  🎉

**Comment by merbla** [comment]
I think we stay with the Visual Studio build as the source of the packages for the moment.  I am not 100% happy with the [build.sh](https://github.com/serilog/serilog/pull/1296/files#diff-0b83f9dedf40d7356e5ca147a077acb4) setup however I am happy not to do a `dotnet pack` in the nix build.

File: build.sh [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
+dotnet --list-sdks
 dotnet restore
 
+echo "🤖 Attempting to build..."
 for path in src/**/*.csproj; do
     dotnet build -f netstandard1.0 -c Release ${path}
     dotnet build -f netstandard1.3 -c Release ${path}
 done
 
+echo "🤖 Running tests..."
```
Comment (nblumhardt) [inline build.sh]: 😂

**Pull request merged**